apps/apps.c

```diff
@@ -2716,6 +2716,125 @@ BIO *bio_open_owner(const char *filename, int format, int private)
     return NULL;
 }
 
+/*
+ * Open a file for writing, owner-read-only. The file is written in
+ * a 'write-to-temp-file-then-rename' basis.
+ */
+#define RND_SIZE 16
+
+static const BIO_METHOD *BIO_f_tempfile(void);
+
+static void make_temp_filename(const char *orig, char *filename)
+{
+    int i, size = RND_SIZE;
+    char *p;
+    char set[] = "0123456789abcdefghijklmnopqrstuvwxyz"
+                 "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+    strcpy(filename, orig);
+    p = filename + strlen(orig);
+    *p++ = '.';
+
+    srand(time(NULL));
```

Comment:
`srand(time(NULL) ^ getpid())` is a common one to try to avoid two processes stomping on each other.  I'm not sure if `getpid()` is universally available.

I think this is moot because a loop that checks the existence by opening the file exclusively won't ever clash.